diff -ruN a/Makefile.am b/Makefile.am
--- a/Makefile.am	2021-05-28 12:38:32.670896220 +0200
+++ b/Makefile.am	2021-05-26 15:47:39.408648474 +0200
@@ -5,8 +5,6 @@
                 src/sscep.h \
                 src/conf.h \
                 src/cmd.h \
-                src/getopt.c \
-                src/getopt.h \
                 src/configuration.c \
                 src/configuration.h \
                 src/engine.c \
diff -ruN a/src/sscep.h b/src/sscep.h
--- a/src/sscep.h	2021-05-28 12:38:32.682899702 +0200
+++ b/src/sscep.h	2021-05-26 15:55:41.295127666 +0200
@@ -17,7 +17,7 @@
 #include <ctype.h>
 #include <signal.h>
 #include <setjmp.h>
-#include "getopt.h"
+#include <getopt.h>
 #include "fileutils_capi.h"
 #include "configuration.h"
 #include "engine.h"
