diff -Naur micropython-1.15/py/formatfloat.c micropython-phoenix/py/formatfloat.c
--- micropython-1.15/py/formatfloat.c	2021-04-14 07:35:31.000000000 +0200
+++ micropython-phoenix/py/formatfloat.c	2021-04-28 11:41:33.452512193 +0200
@@ -90,9 +90,9 @@
 #define FPROUND_TO_ONE 0.999999999995
 #define FPDECEXP 256
 #define FPMIN_BUF_SIZE 7 // +9e+199
-#define fp_signbit(x) signbit(x)
-#define fp_isnan(x) isnan(x)
-#define fp_isinf(x) isinf(x)
+#define fp_signbit(x) ((0 < x) - (x < 0))
+#define fp_isnan(x) __builtin_isnan(x)
+#define fp_isinf(x) __builtin_isinf(x)
 #define fp_iszero(x) (x == 0)
 #define fp_isless1(x) (x < 1.0)
 
@@ -128,7 +128,7 @@
         }
         return buf_size >= 2;
     }
-    if (fp_signbit(f) && !fp_isnan(f)) {
+    if (fp_signbit(f) < 0 && !fp_isnan(f)) {
         *s++ = '-';
         f = -f;
     } else {
diff -Naur micropython-1.15/py/mkenv.mk micropython-phoenix/py/mkenv.mk
--- micropython-1.15/py/mkenv.mk	2021-04-14 07:35:31.000000000 +0200
+++ micropython-phoenix/py/mkenv.mk	2021-04-14 13:27:45.000000000 +0200
@@ -11,6 +11,7 @@
 
 THIS_MAKEFILE := $(lastword $(MAKEFILE_LIST))
 TOP := $(patsubst %/py/mkenv.mk,%,$(THIS_MAKEFILE))
+DEBUG = 1
 
 # Turn on increased build verbosity by defining BUILD_VERBOSE in your main
 # Makefile or in your environment. You can also use V=1 on the make command
@@ -46,6 +47,7 @@
 TOUCH = touch
 PYTHON = python3
 
+CROSS_COMPILE = i386-pc-phoenix-
 AS = $(CROSS_COMPILE)as
 CC = $(CROSS_COMPILE)gcc
 CXX = $(CROSS_COMPILE)g++
diff -Naur micropython-1.15/py/modbuiltins.c micropython-phoenix/py/modbuiltins.c
--- micropython-1.15/py/modbuiltins.c	2021-04-14 07:35:31.000000000 +0200
+++ micropython-phoenix/py/modbuiltins.c	2021-04-14 13:27:45.000000000 +0200
@@ -38,6 +38,7 @@
 #if MICROPY_PY_BUILTINS_FLOAT
 #include <math.h>
 #endif
+#define nearbyint(x) round(x)
 
 #if MICROPY_PY_IO
 extern struct _mp_dummy_t mp_sys_stdout_obj; // type is irrelevant, just need pointer
diff -Naur micropython-1.15/py/modmath.c micropython-phoenix/py/modmath.c
--- micropython-1.15/py/modmath.c	2021-04-14 07:35:31.000000000 +0200
+++ micropython-phoenix/py/modmath.c	2021-04-14 13:27:45.000000000 +0200
@@ -26,6 +26,28 @@
 
 #include "py/builtin.h"
 #include "py/runtime.h"
+#include "lib/libm/libm.h"
+#include "lib/libm_dbl/libm.h"
+#include "lib/libm_dbl/expm1.c"
+#include "lib/libm_dbl/log1p.c"
+#include "lib/libm_dbl/asinh.c"
+#include "lib/libm_dbl/acosh.c"
+#include "lib/libm_dbl/atanh.c"
+#include "lib/libm_dbl/erf.c"
+#include "lib/libm_dbl/tgamma.c"
+#include "lib/libm_dbl/lgamma.c"
+#include "lib/libm_dbl/copysign.c"
+
+
+#ifndef isnan
+#define isnan(x) __builtin_isnan(x)
+#endif
+#ifndef isinf
+#define isinf(x) __builtin_isinf(x)
+#endif
+#ifndef isfinite
+#define isfinite(x) __builtin_isfinite(x)
+#endif
 
 #if MICROPY_PY_BUILTINS_FLOAT && MICROPY_PY_MATH
 
diff -Naur micropython-1.15/py/mpconfig.h micropython-phoenix/py/mpconfig.h
--- micropython-1.15/py/mpconfig.h	2021-04-14 07:35:31.000000000 +0200
+++ micropython-phoenix/py/mpconfig.h	2021-04-14 13:27:45.000000000 +0200
@@ -1600,7 +1600,7 @@
 #endif
 
 #ifndef MP_SSIZE_MAX
-#define MP_SSIZE_MAX SSIZE_MAX
+#define MP_SSIZE_MAX 9223372036854775807
 #endif
 
 // printf format spec to use for mp_int_t and friends
diff -Naur micropython-1.15/py/mpstate.h micropython-phoenix/py/mpstate.h
--- micropython-1.15/py/mpstate.h	2021-04-14 07:35:31.000000000 +0200
+++ micropython-phoenix/py/mpstate.h	2021-04-14 13:27:45.000000000 +0200
@@ -36,6 +36,8 @@
 #include "py/objlist.h"
 #include "py/objexcept.h"
 
+#define nan(x) NAN
+
 // This file contains structures defining the state of the MicroPython
 // memory system, runtime and virtual machine.  The state is a global
 // variable, but in the future it is hoped that the state can become local.
diff -Naur micropython-1.15/py/objcomplex.c micropython-phoenix/py/objcomplex.c
--- micropython-1.15/py/objcomplex.c	2021-04-14 07:35:31.000000000 +0200
+++ micropython-phoenix/py/objcomplex.c	2021-04-14 13:27:45.000000000 +0200
@@ -36,6 +36,10 @@
 #include <math.h>
 #include "py/formatfloat.h"
 
+#ifndef isnan
+#define isnan(x) __builtin_isnan(x)
+#endif
+
 typedef struct _mp_obj_complex_t {
     mp_obj_base_t base;
     mp_float_t real;
diff -Naur micropython-1.15/py/objfloat.c micropython-phoenix/py/objfloat.c
--- micropython-1.15/py/objfloat.c	2021-04-14 07:35:31.000000000 +0200
+++ micropython-phoenix/py/objfloat.c	2021-04-14 13:27:45.000000000 +0200
@@ -31,10 +31,21 @@
 
 #include "py/parsenum.h"
 #include "py/runtime.h"
+#ifndef isinf
+#define isinf(x) __builtin_isinf(x)
+#endif
+#ifndef isnan
+#define isnan(x) __builtin_isnan(x)
+#endif
+#ifndef copysign
+#define copysign(x, y) y == 0.0 ? fabs(x) : fabs(x)*y/fabs(y)
+#endif
+#ifndef signbit
+#define signbit(x) ((0 < x) - (x < 0))
+#endif
 
 #if MICROPY_PY_BUILTINS_FLOAT
 
-#include <math.h>
 #include "py/formatfloat.h"
 
 #if MICROPY_OBJ_REPR != MICROPY_OBJ_REPR_C && MICROPY_OBJ_REPR != MICROPY_OBJ_REPR_D
diff -Naur micropython-1.15/py/parsenum.h micropython-phoenix/py/parsenum.h
--- micropython-1.15/py/parsenum.h	2021-04-14 07:35:31.000000000 +0200
+++ micropython-phoenix/py/parsenum.h	2021-04-14 13:27:45.000000000 +0200
@@ -26,6 +26,7 @@
 #ifndef MICROPY_INCLUDED_PY_PARSENUM_H
 #define MICROPY_INCLUDED_PY_PARSENUM_H
 
+#include <math.h>
 #include "py/mpconfig.h"
 #include "py/lexer.h"
 #include "py/obj.h"
diff -Naur micropython-1.15/py/emitnative.c micropython-phoenix/py/emitnative.c
--- micropython-1.15/py/emitnative.c	2021-04-18 16:27:07.000000000 +0200
+++ micropython-phoenix/py/emitnative.c	2021-05-10 15:28:25.395437782 +0200
@@ -2396,7 +2396,7 @@
         } else if (op == MP_BINARY_OP_MULTIPLY) {
             ASM_MUL_REG_REG(emit->as, REG_ARG_2, reg_rhs);
             emit_post_push_reg(emit, vtype_lhs, REG_ARG_2);
-        } else if (MP_BINARY_OP_LESS <= op && op <= MP_BINARY_OP_NOT_EQUAL) {
+        } else if (MP_BINARY_OP_LESS == op && op <= MP_BINARY_OP_NOT_EQUAL) {
             // comparison ops are (in enum order):
             //  MP_BINARY_OP_LESS
             //  MP_BINARY_OP_MORE
 